### ISSUE TITLE
Failed CP's can break our CI/CD

### ISSUE THREAD

**Issue by roryabraham** [Expensify/App#6863]
## Action Performed:
1. Merge a dummy PR.
1. Create a revert PR for that dummy PR.
1. Give the revert PR the `CP Staging` label.
1. Merge the revert PR.
1. You'll observe that the app version on the deploy checklist increases. (BUG)
1. There _should_ be a merge conflict with the CP PR.
1. Close the CP PR without resolving conflicts or merging it.
1. Then _without running any other CP's first_, close the checklist to run a production deploy.

## Expected Result:
The production and staging deploys should complete successfully, and we should have a new checklist.

## Actual Result:
The production and staging deploys complete successfully, but creating a new checklist fails with the following type of error:

```
Found tag of previous StagingDeployCash: 1.1.22-1
Getting pull requests merged between the following refs: 1.1.22-1 1.1.22-2
Running command:  git log --format="{[%B]}" 1.1.22-1...1.1.22-2
fatal: ambiguous argument '1.1.22-1...1.1.22-2': unknown revision or path not in the working tree.
```

Where in this case `1.1.22-1` is the tag that was posted in the deploy checklist, but never actually created.

## Workaround:
**DANGER:**
1. Manually update the checklist to have the tag matching the latest production deploy (GitHub Release).
1. Re-run the `finishReleaseCycle.yml` workflow
1. **IMPORTANT NOTE:** This will attempt to deploy all the code from staging to production. It's _imperative_ that you manually kill the workflow before that happens. It's best to kill the `updateProtectedBranch` workflow that's attempting to merge the code from `staging` -> `production`, but if you're too slow then you can kill the `deploy.yml` or `platformDeploy.yml` as well.

## Platform:
GitHub Actions

**Comment by roryabraham** [comment]
We've seen similar problem that suggests a similar solution to this, namely: "Only update the app version on the `StagingDeployCash` once the tag has been created _and_ pushed"

**Comment by roryabraham** [comment]
No update here

**Comment by roryabraham** [comment]
Another good improvement to make once https://github.com/pascalgn/automerge-action/pull/177 is merged would be to delete the unmerged PR branch if the automerge-action fails.

**Comment by roryabraham** [comment]
No update

**Comment by roryabraham** [comment]
No update

**Comment by roryabraham** [comment]
This is a pretty rare edge case that requires some human error to reproduce it (i.e: begin a CP but don't finish it). Going to move this to monthly and send it back to the pool for now. Would still be good to fix someday.

**Comment by roryabraham** [comment]
Maybe worth doing in concert with https://github.com/Expensify/Expensify/issues/195693, reopening for now. cc @Expensify/mobile-deployers in case anyone wants to work on this. It's a good edge case to be aware of, especially since the fix is sort of high-risk / you have to be careful not to accidentally deploy a bunch of un-QA'd code.

**Comment by puneetlath** [comment]
This isn't a bug that a user would experience, so I'm going to slap the `NewFeature` label on it and unassign.

**Comment by JmillsExpensify** [comment]
Since this is a new feature and no one is assigned, I'm going to close the issue. Anyone that wants to re-open please assign yourself when doing so.

**Comment by roryabraham** [comment]
This is a a bug, not a new feature.

> This isn't a bug that a user would experience

the "user" in this case is a deployer

**Comment by thienlnam** [comment]
@roryabraham I've started to look into this ([Slack Thread](https://expensify.slack.com/archives/C07J32337/p1675366658928589?thread_ts=1675350839.930089&cid=C07J32337))

I am a little unsure how to proceed since this [cherry pick action](https://github.com/Expensify/App/actions/workflows/cherryPick.yml) fails when there is some kind of merge conflict and needs to be merged manually.

But this workflow only breaks when we abort the CP and not even merge manually right?  So if we were to move the logic to bump the version in the checklist to after it was successful or when the manual PR was merged - would it even be possible to check that it was a manual merge?
